src/promised-singleton.ts

```diff
@@ -46,19 +46,18 @@ export class PromisedSingleton<T> {
   }
 
   private async generateSingletonAndCache(): Promise<T> {
-    return this.generator.then(response => {
-      this.cachedResponse = response;
-      return response;
-    });
+    const result = await this.generator();
+    this.cachedResponse = result;
```

Comment:
usage Q: in order to bust cache and re-run, one just simply re-runs the `generator`?

Reply:
Since the generator is a `Promise`, it can only be executed once. Generally with singletons, you only want them to be generated once, but if you do need to generate a new one, I would just create a new `PromisedSingleton` at the source.